### PR TITLE
[#19] [Integrate] As a user, I can see many question types when doing survey feedback

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/luongvo/flutter-survey/issues/19

## What happened 👀

The list of questions must be implemented based on the API response:
- Click `Close` button to close survey detail screen. 
- Each question type has a different layout and UI which must be respected.
- Allow users to select a response and store the selected responses.
- Click the `next` button will save the selection and navigate to the next question with a horizontal swipe gesture.
 
## Insight 📝

- We need to [add missing `Answer` data from `Question` list response](https://github.com/luongvo/flutter-survey/pull/65/files#diff-bc99db5a79199a2ad907e7560ee727719f7b6b0ea071dc5ba12520d070fa5cfdR16).
- Convert `Question.displayType` data type from String to enum by [enumFromString](https://github.com/luongvo/flutter-survey/pull/65/files#diff-c149f41c10686502d0127f712053c31429cb39941138d01e2cd62640c2307970R56-R59) helper.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/179395808-bbebc0b7-ef1e-498e-a53d-9d303239e917.mp4

### CODE REVIEW

File: lib/models/question.dart [hunk 1]
```diff
@@ -28,28 +32,44 @@ class Question extends Equatable {
...
+        answers,
       ];
 
   factory Question.fromQuestionResponse(QuestionResponse response) {
     return Question(
       id: response.id,
       text: response.text ?? '',
       displayOrder: response.displayOrder ?? 0,
-      displayType: response.displayType ?? '',
+      displayType: enumFromString(DisplayType.values, response.displayType) ??
```
Comment (markgravity) [inline lib/models/question.dart:44]: Why do we use [JsonEnum](https://pub.dev/packages/json_serializable#enums)? 🤔

**Review by markgravity — COMMENTED** [review]
Please check the leading zero on the double numbers 🙏

File: lib/models/question.dart (hunk 1 above)
Comment (luongvo) [reply lib/models/question.dart]: @markgravity you're right, fixed by https://github.com/luongvo/flutter-survey/pull/65/commits/7040e97814651b590b0f1c7044342c5b4cb58ffb

**Comment by luongvo** [comment]
@doannimble @manh-t the last one for this release :pray:

**Comment by manh-t** [comment]
There is an overflow issue when opening the keyboard. Rest LGTM

**Comment by luongvo** [comment]
> There is an overflow issue when opening the keyboard.

Fixed by https://github.com/luongvo/flutter-survey/pull/65/commits/88f8fcd4c08159baee69854b635a32195add55e7

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**